sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -422,12 +422,30 @@ case class DataSourceStrategy(conf: SQLConf) extends Strategy with Logging with
 }
 
 object DataSourceStrategy {
+
+  /**
+   * Tries to translate a Catalyst [[Expression]] into data source [[Expression]].
+   *
+   * @return a `Some[catalog.v2.expressions.Expression]` if the input [[Expression]]
+   *         is convertible, otherwise a `None`.
+   */
+  protected[sql] def translateExpression(
+      value: Expression): Option[catalog.v2.expressions.Expression] = {
+    // TODO: currently we only support literal. Will add more public expression in future.
+    value match {
+      case l: Literal =>
+        Some(catalog.v2.expressions.LiteralValue(l.value, l.dataType))
+      case _ =>
+        None
+    }
+  }
+
   /**
    * The attribute name of predicate could be different than the one in schema in case of
    * case insensitive, we should change them to match the one in schema, so we do not need to
    * worry about case sensitivity anymore.
    */
-  protected[sql] def normalizeFilters(
+  protected[sql] def normalizeAttrNames(
```

Comment:
This rename results in a lot of additional file changes. Can we avoid renaming?

Reply:
If we don't, then it's weird to use `normalizeFilters` to normalize update expressions.

Reply:
Yes. I'll open another pr for the renaming.